docs/src/main/asciidoc/getting-started-testing.adoc

```diff
@@ -1506,3 +1506,94 @@ For `@QuarkusIntegrationTest` tests that result in launcher the application as a
 This can be used by `QuarkusTestResourceLifecycleManager` that need to launch additional containers that the application will communicate with.
 ====
 
+== Testing Components
+
+In Quarkus, the component model is built on top CDI.
+Therefore, Quarkus provides the `QuarkusComponentTest`, a JUnit extension to ease the testing of components and mocking of their dependencies.
+This extension is available in the `quarkus-junit5-component` dependency.
+
+Let's have a component `Foo`:
+
+[source, java]
+----
+package org.acme;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+@ApplicationScoped <1>
+public class Foo {
+
+    @Inject
+    Charlie charlie; <2>
+
+    public String ping() {
+        return charlie.ping();
+    }
+}
+----
+<1> `Foo` is an `@ApplicationScoped` CDI bean.
+<2> `Foo` depends on `Charlie` which declares a method `ping()`.
+
+Then a component test could look like:
+
+[source, java]
+----
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mockito.Mockito;
+import org.acme.Charlie;
+import org.acme.Foo;
+
+public class FooTest {
+
+    @RegisterExtension <1>
+    static final QuarkusComponentTest test = new QuarkusComponentTest(Foo.class);
+
+    @Inject
+    Foo foo; <2>
+
+    // Inject a mock that is created automatically...
+    @Inject
+    Charlie charlie; <3>
+
+    @Test
+    public void testPing() {
+        Mockito.when(charlie.ping()).thenReturn("OK"); <4>
+        assertEquals("OK", foo.ping());
+    }
+}
+----
+<1> The `QuarkusComponentTest` extension is configured in a static field.
+<2> The test injects the component under the test.
+<3> The test also injects `Charlie`, a dependency for which a synthetic `@Singleton` bean is registered automatically. The injected reference is an "unconfigured" Mockito mock. 
+<4> We can leverage the Mockito API in a test method to configure the behavior.
+
+=== Lifecycle
+
+So what exactly does the `QuarkusComponentTest` do?
+It starts the CDI container and registers a dedicated `SmallRyeConfig` during the `before all` test phase. 
+The container is stopped and the config is released during the `after all` test phase.
+The fields annotated with `@Inject` are injected after a test instance is created and unset before a test instance is destroyed.
+Finally, the CDI request context is activated and terminated per each test method.
+
+=== Auto Mocking Unsatisfied Dependencies
+
+Unlike in regular CDI environments the test does not fail if a component injects an unsatisfied dependency. 
+Instead, a synthetic bean is registered automatically for each combination of required type and qualifiers of an injection point that resolves to an unsatisfied dependency.
```

Comment:
I don't think `synthetic` means anything to users

Reply:
Well, yeah but I'd still prefer to be clear that the registered bean is not a regular class-based bean. And it's also consistent with other parts of the CDI docs (https://quarkus.io/guides/cdi#ok-you-said-that-there-are-several-kinds-of-beans, https://quarkus.io/guides/cdi-integration#synthetic_beans, etc.).

Reply:
Again, that doesn't mean anything to users. Sure, it's technically correct, but it's a term that very few people know, so it's not as helpful as it could be

Reply:
It might not be that helpful but it does not hurt IMO :shrug: